### PR TITLE
Network.Wai.Test: Add support for source locations to assertion primitives

### PR DESCRIPTION
**Pull request by sol**
…tives

Before submitting your PR, check that you've:

- [x] Bumped the version number
- [x] Documented new APIs with [Haddock markup](https://www.haskell.org/haddock/doc/html/index.html)
- [x] Added [`@since` declarations](http://haskell-haddock.readthedocs.io/en/latest/markup.html#since) to the Haddock

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [ ] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

File: wai-extra/wai-extra.cabal [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Name:                wai-extra
-Version:             3.0.32
+Version:             3.1.0
```
Comment (snoyberg) [inline wai-extra/wai-extra.cabal:2]: Please include a lower bound on base for 4.10 here to prevent untested GHC versions from being used.

Comment (sol) [reply wai-extra/wai-extra.cabal:2]: Done. I cherry-picked 38a828a500ae0bc5e0f6210e0cdb3f2cdeb0d31d.

Comment (snoyberg) [reply wai-extra/wai-extra.cabal:2]: That's not what I meant. I still want @kazu-yamamoto's sign-off before introducing those lower bounds on the rest of the packages here

Comment (sol) [reply wai-extra/wai-extra.cabal:2]: Ok, just for `wai-extra` now.

**Review by snoyberg — APPROVED** [review]
Thanks! Sorry for all the extra legwork here

**Pull request merged**